### DISCUSSION TITLE
Web cam stream which requires both HTTPS and basic authorization-header

### DISCUSSION THREAD

**Discussion by joriws** [blakeblackshear/frigate#3975] [Ask A Question] [answered]
I have TP-Link KC120 and with curl I can stream video/pictures to ffmpeg which writes them to disc.

`
**curl** -vv -k -u xxxxx:yyyyy --ignore-content-length "https://192.168.0.17:19443/https/stream/mixed?video=H264&audio=G711" --output - | **ffmpeg** -y -i - test.mp4

root@tensor:~/frigate# ffprobe test.mp4
ffprobe version 4.3.4-0+deb11u1 Copyright (c) 2007-2021 the FFmpeg developers
  built with gcc 10 (Debian 10.2.1-6)
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'test.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    encoder         : Lavf58.45.100
  Duration: 00:01:15.88, start: 0.000000, bitrate: 971 kb/s
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuvj420p(pc), 1920x1080, 968 kb/s, 25 fps, 25 tbr, 12800 tbn, 50 tbc (default)
    Metadata:
      handler_name    : VideoHandler
`

also test.mp4 outputs fine and I can view it with VLC.

**My question is that how do I enter above as config.yml for Frigate?**

- I don't think ffmpeg supports http**s**
- I don't think with ffmpeg I can add basic authorization -header to https call
- rtsp simple server does not support https input to convert live to rtsp/rtmp/hls
- motion does not support https input (motion has rtmp streaming)
- test.mp4 - can I use that somehow as FIFO. I think the codec parameters of mp4 stream etc will be problem as if they are only on start of file/fifo.
- Frigate is enhanced with better stdin-streaming and pre-stream command which runs curl?

Any other ideas/suggestions/solutions?

**Comment by NickM-27** [comment, marked as answer]
1. Look at https://github.com/AlexxIT/go2rtc
2. Ffmpeg does support https https://ffmpeg.org/ffmpeg-protocols.html#Icecast

**Reply by joriws** [reply]
@NickM-27 thank you for good tips. 1st tip - checking it if it supports https header setting.. About second bullet, nice, documentation just is missing - in 3.14 http just reconnect_on_network_error mentions TLS. Maybe if ffmpeg http-input "headers" option could be utilized/encoded to ffmpeg input_args etc to Frigate, maybe then it could handle directly..

**Reply by NickM-27** [reply]
The entire ffmpeg command is customizable in frigate so whatever is possible in ffmpeg is possible in frigate

**Reply by joriws** [reply]
I think I need curl in front - or it is simpler as how TPLINK KC120 is sending video out. With go2rtc and it's exec-functionality and shell script I was able to solve the problem, and best of all it takes 1% of CPU. So thank you @NickM-27 for sharing go2rtc, I was not aware of such software, but it seems to be quite new. 

ref: https://github.com/AlexxIT/go2rtc/issues/64#

**Reply by NickM-27** [reply]
Glad that is working for you
